src/main/java/zmq/Mailbox.java

```diff
@@ -89,7 +89,7 @@ public Command recv(long timeout)
         //  Wait for signal from the command sender.
         boolean rc = signaler.waitEvent(timeout);
         if (!rc) {
-            assert (errno.get() == ZError.EAGAIN || errno.get() == ZError.EINTR);
+            assert (errno.get() == ZError.EAGAIN || errno.get() == ZError.EINTR) : errno.get();
```

Comment:
I suppose there might be some tiny possibility that `errno` is reset in another thread while this line is being executed, so perhaps we should read the value one time only? i.e.

```java
int errno = errno.get();
assert (errno == ZError.EAGAIN || errno == ZError.EINTR) : errno;
```

Reply:
I'm pretty confident that it cannot happen: Errno holds a ThreadLocal variable, its value is per-thread.